### PR TITLE
Fix bugs in `process_j312.sh`

### LINKED ISSUE #99: [Bug]: bugs in `process_j312.sh`
## Expected behavior
The [`process_j312.sh`](https://github.com/arfc/saltproc/blob/75bd4097345d55604b2b5712d0f3ed7bf94b03aa/scripts/xsdata/process_j312.sh) script should run without any errors, and the `.xsdir` file it produces should not throw any errors when using the `xsdirconvert.pl` script detailed [here](https://serpent.vtt.fi/mediawiki/index.php/Installing_and_running_Serpent#Setting_up_the_data_libraries)

## Describe the bug
1. Instruction on line 90 causes an error with the `mv` command
2. The `sss_jeff312.xsdir` file created by the script assumes that the `acedata` directory is stored under a `jeff312` directory. This can lead to errors when trying to use `xsdircinvert.pl`.
3. There is an extra `directory` line in the `sss_jeff312.xsdir` file created by the script.

Additionally, it also seems that the script does not download the decay chains from JEFF 3.3, despite this being necessary for the tests.

## To Reproduce
1. Download and run the script to see the first error
2. Delete the `>` on line 90 and rerun the script. Use the `xsdirconvert.pl` script (see link above) to see the third error.
3. Delete `/jeff312`, from line 34 and rerun a third time. Use the `xsdirconvert.pl` script (see link above) to see the third error.

## Supporting Informatioon
If applicable, add program output with triple backticks, a gist of your output, or a screenshot to help explain your problem.

### Bug 1:
```
.
.
.
  inflating: /home/ooblack/test/900/Zr96.dir  
mv: missing destination file operand after 'temp.ACE'
Try 'mv --help' for more information.
.
.
.
```

### Bug 2:
```
$ perl xsdirconvert.pl sss_jeff312.xsdir > sss_jeff312.xsdata
89225.09c: ACE file "/home/ooblack/test/jeff312/acedata/Ac225-900.ACE" does not exist.
.
.
.
```

### Bug 3:
``` 
$ perl xsdirconvert.pl sss_jeff312.xsdir > sss_jeff312.xsdata
directory: Data type 0 not supported.

```

## System Specifications:
 
 - Software version: Saltproc v0.3.0
 - Platform/OS: Ubuntu 18.04 and 21.04

## How can this issue be closed?
This issue can be closed when a PR exists that fixes all three bugs

### PR DESCRIPTION
**Pull request by yardasol**
## Summary of changes

This PR fixes the bugs described in #99 
This PR also renames `process_j312.sh` to `process_j312.bash` to clarify that the script uses `bash` syntax, and will not work properly (or even fail) when run as a native shell script.
This PR also adds detection of existing compressed cross section library files, neutron data files, and decay chain file to reduce time it takes to rerun the script by skipping the download commands.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Required for Merging
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
  - [x] CI tests pass
  - [x] Local tests pass (including Serpent2 integration tests)

## Associated Issues and PRs

- Issue: closes #99 

## Associated Developers

- Dev: @munkm 

## Checklist for Reviewers

Reviewers should use [this link](https://arfc.github.io/manual/guides/pull_requests) to get to the 
Review Checklist before they begin their review.

### CODE REVIEW

**Comment by yardasol** [comment]
block merging until #103 is resolved.

File: scripts/xsdata/process_j312.bash [hunk 1]
```diff
@@ -17,29 +19,47 @@ TEMPS=(900)
...
+then
+    wget -O $DATADIR/ACEs_THERMK.zip $LN$SLUG1
+fi
+
+
+# Get neutron induced and spontaneous fission yield data from JEFF 3.3
+if [[ ! -f $DATADIR/sss_jeff33.nfy ]]
+then
+    wget -O $DATADIR/sss_jeff33.nfy https://www.oecd-nea.org/dbdata/jeff/jeff33/downloads/JEFF33-nfy.asc
+fi
```
Comment (abachma2) [inline scripts/xsdata/process_j312.bash:39]: Does this pull data from JEFF 3.3? I thought this was for getting data from JEFF 3.1.2 ???

**Review by abachma2 — COMMENTED** [review]
Overall this looks good. I did try this out and it does work well. I just have the one question about JEFF versions.

File: scripts/xsdata/process_j312.bash (hunk 1 above)
Comment (yardasol) [reply scripts/xsdata/process_j312.bash:39]: There is no spontaneous or delayed neutron fission data in JEFF 3.1.2, nor is there decay data. The data from the JEFF 3.3 library is backwards compatible with the JEFF 3.1.2 library.

Comment (abachma2) [reply scripts/xsdata/process_j312.bash:39]: That makes sense. Just wanted to clarify the reason for the version change.

**Review by abachma2 — APPROVED** [review]
I approve this pull request. I will wait to merge until #103 is resolved.

**Pull request merged**